lib/queue_monitor.ts

```diff
@@ -0,0 +1,55 @@
+import * as winston from "winston";
+
+import { TelemetryClient } from "applicationinsights";
+import { IContext } from "azure-functions-types";
+import { createQueueService } from "azure-storage";
+import { MESSAGE_QUEUE_NAME } from "./created_message_queue_handler";
+import { EMAIL_NOTIFICATION_QUEUE_NAME } from "./emailnotifications_queue_handler";
+import { getRequiredStringEnv } from "./utils/env";
+import { configureAzureContextTransport } from "./utils/logging";
+import { WEBHOOK_NOTIFICATION_QUEUE_NAME } from "./webhook_queue_handler";
+
+const queueConnectionString = getRequiredStringEnv("QueueStorageConnection");
+const queueService = createQueueService(queueConnectionString);
+
+// Whether we're in a production environment
+const isProduction = process.env.NODE_ENV === "production";
+
+const appInsightsClient = new TelemetryClient();
+
+/**
+ * A function to store the length of Azure Storage Queues
+ * into Application Insights Metrics.
+ *
+ * To query these values in the Analytics panel type:
+ *
+ * customMetrics
+ *   | where * has  "queue.length"
+ *   | order by timestamp desc
+ */
+export function index(context: IContext): void {
+  const logLevel = isProduction ? "info" : "debug";
+  configureAzureContextTransport(context, winston, logLevel);
+
+  [
```

Comment:
would be nice to compose this list automatically, is there a way to enumerate the queues by querying the queues API?

Reply:
I see, never mind then :)